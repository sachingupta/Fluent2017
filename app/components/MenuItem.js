import React from 'react';
import { StyleSheet, View, Text, Image, Dimensions } from 'react-native';

const { width } = Dimensions.get('window');

export default MenuItem = ({
  source,
  title,
  description,
  price,
}) => (
  <View>
    <Image
      resizeMode='contain'
      source={source}
      style={styles.image}
    />
    <Text style={styles.title}>{title}</Text>
    <Text style={styles.description}>{description}</Text>
    <Text style={styles.price}>{price}</Text>
  </View> 
)

const styles = StyleSheet.create({
  title: {
    fontFamily: 'AlegreyaSansSC-Light',
    fontSize: 26,
    marginTop: 6,
    marginLeft: 10,
    marginBottom: -4,
  },
  description: {
    fontFamily: 'AlegreyaSansSC-Regular',
    fontSize: 14,
    marginLeft: 10,
  },
  image: {
    height: 220,
    width,
  },
  price: {
    fontFamily: 'AlegreyaSansSC-Medium',
    fontSize: 28,
    marginLeft: 10,
    marginBottom: 10,
  }
})