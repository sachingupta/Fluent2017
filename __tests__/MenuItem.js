import 'react-native';
import React from 'react';
import MenuItem from '../app/components/MenuItem.js';

import renderer from 'react-test-renderer';

/*
* MenuItem props:
* source function or object
* title string
* description string
* price string
*/

it('renders correctly', () => {
  const tree = renderer.create(
    <MenuItem
      title='Guacamole'
      description='Lorem ipsum'
      price='$4.99'
      source={require('../app/assets/images/guacamole.jpg')}
    />
  );
  expect(tree).toMatchSnapshot();
});
