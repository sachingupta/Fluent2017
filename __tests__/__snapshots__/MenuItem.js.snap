// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": 220,
        "width": 750,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "AlegreyaSansSC-Light",
        "fontSize": 26,
        "marginBottom": -4,
        "marginLeft": 10,
        "marginTop": 6,
      }
    }
  >
    Guacamole
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "AlegreyaSansSC-Regular",
        "fontSize": 14,
        "marginLeft": 10,
      }
    }
  >
    Lorem ipsum
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "AlegreyaSansSC-Medium",
        "fontSize": 28,
        "marginBottom": 10,
        "marginLeft": 10,
      }
    }
  >
    $4.99
  </Text>
</View>
`;
